webapp/src/main/java/io/github/microcks/util/grpc/GrpcMetadataUtil.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The Microcks Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.microcks.util.grpc;
+
+import io.github.microcks.util.script.HttpHeadersStringToStringsMap;
+import io.github.microcks.util.script.StringToStringsMap;
+import io.grpc.CallOptions;
+import io.grpc.Context;
+import io.grpc.Metadata;
+
+import java.io.UnsupportedEncodingException;
+import java.util.*;
+
+/**
+ * Helper class containing utility to deal with GrpcMetadata.
+ */
+public class GrpcMetadataUtil {
+
+   /* Key used to pass gRPC metadata from interceptor to server via context */
+   public static final Context.Key<Metadata> METADATA_CTX_KEY = Context.key("grpc-metadata");
+
+   private GrpcMetadataUtil() {
+      // Private constructor to hide the implicit public one.
+   }
+
+   /**
+    * Convert Metadata sent with a gRPC requests into a map of key-value pairs with metadata key to list of values.
+    * 
+    * @param metadata The gRPC Metadata to convert
+    * @return A StringToStringsMap containing all Metadata key-value pairs
+    * @throws UnsupportedEncodingException If metadata contains invalid binary value which could not be decoded into a
+    *                                      string.
+    */
+   public static StringToStringsMap convertToMap(Metadata metadata) throws UnsupportedEncodingException {
+      StringToStringsMap result = new HttpHeadersStringToStringsMap();
+
+      Set<String> keys = metadata.keys();
+      for (String key : keys) {
+         // Depending on the suffix of the key we either extract binary values or
+         // ASCII string values.
+         if (key.endsWith("-bin")) {
```

Comment:
Is it a common convention? Don't we have another way of telling how a key should be marshalled/unmarshalled? 
Plus: shouldn't we have at least a constant definition like `BINARY_VALUE_SUFFIX`?

Reply:
Yes, this is a common convention and actually required by grpc.
Here a snippet from the [grpc-specification](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md):

> Applications define binary headers by having their names end with "-bin". Runtime libraries use this suffix to detect binary headers and properly apply base64 encoding & decoding as headers are sent and received.


The constant was a great suggestion. Actually,`io.grpc.Metadata` defines already the constant BINARY_HEADER_SUFFIX which we can use.

When iterating over all keys in the metadata, we only get the name of the key and not the type of the key . There is one alternative, though. We could replace it by checking explicitly with which marshaller a key exists:
```

if (metadata.containsKey(Metadata.Key.of(key, Metadata.BINARY_BYTE_MARSHALLER) )) {
    // binary metadata
    ....
} else { 
    // ASCII string metadata
    ....
}
```

Reply:
Awesome!